src/main/java/meteordevelopment/meteorclient/settings/AbstractRegistryItemSetting.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * This file is part of the Meteor Client distribution (https://github.com/MeteorDevelopment/meteor-client).
+ * Copyright (c) Meteor Development.
+ */
+
+package meteordevelopment.meteorclient.settings;
+
+import net.minecraft.nbt.NbtCompound;
+import net.minecraft.registry.Registry;
+import net.minecraft.util.Identifier;
+
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+public abstract class AbstractRegistryItemSetting<V> extends Setting<V> {
+    public final Predicate<V> filter;
+    public final Registry<V> registry;
+
+    protected AbstractRegistryItemSetting(String name, String description, V defaultValue, Consumer<V> onChanged, Consumer<Setting<V>> onModuleActivated, IVisible visible, Predicate<V> filter, Registry<V> registry) {
+        super(name, description, defaultValue, onChanged, onModuleActivated, visible);
+
+        this.filter = filter;
+        this.registry = registry;
+    }
+
+    @Override
+    protected final V parseImpl(String str) {
+        return parseId(registry, str);
+    }
+
+    @Override
+    protected final boolean isValueValid(V value) {
+        return filter == null || filter.test(value);
+    }
+
+    @Override
+    public final Iterable<Identifier> getIdentifierSuggestions() {
+        return registry.getIds();
+    }
+
+    @Override
+    protected final NbtCompound save(NbtCompound tag) {
+        tag.putString("value", registry.getId(get()).toString());
+
+        return tag;
+    }
+
+    @Override
+    protected final V load(NbtCompound tag) {
+        value = registry.get(Identifier.of(tag.getString("value", "")));
+
+        if (filter != null && !filter.test(value)) {
+            for (V t : registry) {
```

Comment:
should not fallback to a random valid value from the registry, should fallback to `resetImpl()` instead

Reply:
huh yeah that’s scary why do we do that??
but yeah that’s better, just check if `value` is null